doc/users/cmake_user.html

```diff
@@ -838,32 +852,37 @@ <h2>  <code>deal.IIConfig.cmake</code> </h2>
 
 #
 # Feature configuration: The following booleans are set to "ON" or "OFF" depending
-# on the current feature configuration:
+# on the current feature configuration (note that other variables are defined
+# only if one of the following variable is "ON", for example DEAL_II_MPI_VERSION
+# is only defined if DEAL_II_WITH_MPI is "ON"):
 #
 
 DEAL_II_WITH_64BIT_INDICES
 DEAL_II_WITH_ADOLC
 DEAL_II_WITH_ARPACK
 DEAL_II_WITH_ASSIMP
-DEAL_II_WITH_BOOST
 DEAL_II_WITH_COMPLEX_VALUES
 DEAL_II_WITH_CUDA
 DEAL_II_WITH_GINKGO
-DEAL_II_WITH_GMSH
 DEAL_II_WITH_GSL
+DEAL_II_WITH_GMSH
```

Comment:
I would rather fix the order in `include/deal.II/base/config.h.in` to be alphabetical than here.